projects/plugins/jetpack/extensions/blocks/subscriptions/index.js

```diff
@@ -58,6 +59,28 @@ export const settings = {
 	attributes,
 	edit,
 	save,
+	transforms: {
+		from: [
+			{
+				type: 'block',
+				isMultiBlock: false,
+				blocks: [ 'core/legacy-widget' ],
+				isMatch: ( { idBase, instance } ) => {
+					if ( ! instance?.raw ) {
+						return false;
+					}
+					return idBase === 'blog_subscription';
+				},
+				transform: ( { instance } ) => {
+					return createBlock( 'jetpack/subscriptions', {
+						showSubscribersTotal: instance.raw.show_subscribers_total,
+						submitButtonText: instance.raw.subscribe_button,
```

Comment:
@arcangelini :wave: have you tried changing it to some different custom string! 
Example: `submitButtonText: 'Subscribe to my newsletter'`.
I tried something similar but it is not working for me somehow! :disappointed:  `submitButtonText` value is set to default value that you mentioned in the `attributes.js` file
Can you please try at your end and confirm?
Thanks

Reply:
Yeah, if you look at my last screenshot it shows the placeholder text as "Placeholder" and the submit button as "Button".

I can test again though!

Reply:
I'm trying to understand what this block of code does?
```
return createBlock( 'jetpack/subscriptions', {
    showSubscribersTotal: instance.raw.show_subscribers_total,
    submitButtonText: instance.raw.subscribe_button,
    subscribePlaceholder: instance.raw.subscribe_placeholder,
} );
```
If you see in the screenshot, I have commented default value of `submitButtonText` and if it has to read the value from `index.js` file `instance.raw.subscribe_button` then I have given some hard coded value `custom button value` which is not getting reflected. Button text is empty actually.

![POC](https://user-images.githubusercontent.com/21127788/135686631-7e3b9840-239b-4824-a55b-356b52774158.png)

Reply:
@arcangelini, no need to test it again. It is working as expected. I understand now when that block of code is going to run.
Thank You :smile: